### PR TITLE
Implement changes discussed on Slack

### PR DESCRIPTION
**Pull request by aglavic**
As there were no further comments about my proposed changes of the class attributes on Slack I have implemented these as they have been proposed/discussed to prepare for release on the general assembly.

### CODE REVIEW

File: orsopy/fileio/base.py [hunk 1]
```diff
@@ -468,6 +485,42 @@ class Column(Header):
...
+    error_type: Optional[Literal["uncertainty", "resolution"]] = None
+    distribution: Optional[
+        Tuple[Literal["gaussian", "triangular", "uniform", "lorentzian"], Literal["sigma", "FWHM"]]
+    ] = None
+
+    yaml_representer = Header.yaml_representer_compact
+
+    def to_sigma(self):
+        """
+        Returns the multiplicative factor needed to convert a FWHM to sigma.
```
Comment (arm61) [inline orsopy/fileio/base.py:504]: Is there a reference for these conversions? In particular the triangular and uniform ones?

**Review by arm61 — APPROVED** [review]

File: orsopy/fileio/base.py (hunk 1 above)
Comment (aglavic) [reply orsopy/fileio/base.py:504]: Well, I got them out of my memory and checked with Wikipedia, but I can see if I find a text book to reference here.

Comment (arm61) [reply orsopy/fileio/base.py:621]: I hadn't ever thought about the conversations for triangular and uniform is all. Happy with your memory!

Comment (aglavic) [reply orsopy/fileio/base.py:631]: I added some comments and reference now.

File: orsopy/fileio/data_source.py [hunk 2]
```diff
@@ -32,8 +35,8 @@ class Experiment(Header):
 
     title: str
     instrument: str
-    start_date: str
-    probe: Union[Literal["neutrons", "x-rays"]]
+    start_date: datetime
+    probe: Literal["neutrons", "x-rays"]
```
Comment (bmaranville) [inline orsopy/fileio/data_source.py:39]: The list of probes in Nexus is (from https://manual.nexusformat.org/classes/base_classes/NXsource.html?highlight=probe)

probe: (optional) [NX_CHAR](https://manual.nexusformat.org/nxdl-types.html#nx-char)
type of radiation probe (pick one from the enumerated list and spell exactly)
Any of these values:
- neutron
- x-ray
- muon
- electron
- ultraviolet
- visible light
- positron
- proton

File: orsopy/fileio/base.py [hunk 3]
```diff
@@ -585,6 +605,62 @@ class Column(Header):
...
+@orsodataclass
+class ErrorColumn(Header):
+    """
+    Information about a data column.
+    """
+
+    error_of: str
+    error_type: Optional[Literal["uncertainty", "resolution"]] = None
+    distribution: Optional[
+        Tuple[Literal["gaussian", "triangular", "uniform", "lorentzian"], Literal["sigma", "FWHM"]]
```
Comment (bmaranville) [inline orsopy/fileio/base.py:617]: Grouping the distribution type with the width specifier here seems unecessary.  Why not split out the "sigma" vs "FWHM" to another attribute?

File: orsopy/fileio/data_source.py [hunk 4]
```diff
@@ -61,30 +64,31 @@ class Sample(Header):
...
+    beam, if a spin analyser is present.
+    """
+
+    unpolarized = "unpolarized"
+    p = "p"
+    m = "m"
+    mm = "mm"
+    mp = "mp"
+    pm = "pm"
+    pp = "pp"
```
Comment (bmaranville) [inline orsopy/fileio/data_source.py:87]: There seemed to be general agreement on Slack that we could add ```["?m", "m?", "?p", "p?"]``` to represent half-polarized states (where the part that has unknown/unmeasured polarization is then explicit) - these would replace the bare ```l"p", "m"]``` that don't provide information about which half is unknown.

File: tests/test_example2.ort [hunk 5]
```diff
@@ -63,8 +63,8 @@
 # columns:
 #     - {name: Qz, unit: 1/angstrom, dimension: WW transfer}
 #     - {name: R, dimension: reflectivity}
-#     - {name: sR, dimension: error-reflectivity}
-#     - {name: sQz, unit: 1/angstrom, dimension: resolution-WW transfer}
+#     - {error_of: R}
+#     - {error_of: Qz, error_type: resolution}
```
Comment (bmaranville) [inline tests/test_example2.ort:67]: shouldn't an indication of FWHM vs sigma make it to this column descriptor?

File: orsopy/fileio/base.py (hunk 3 above)
Comment (aglavic) [reply orsopy/fileio/base.py]: Yes, good point. We separated it into "distribution" and "value_is" attributes.

File: orsopy/fileio/data_source.py (hunk 4 above)
Comment (aglavic) [reply orsopy/fileio/data_source.py:87]: Question marks would make it problematic to use as attribute names. Jochen and I preferred to go with "o" instead, hope that's OK.

File: orsopy/fileio/data_source.py (hunk 2 above)
Comment (aglavic) [reply orsopy/fileio/data_source.py:39]: Replaced neutrons and x-rays with the values here. We don't see the use of the other radiation types at the moment.

File: orsopy/fileio/data_source.py (hunk 4 above)
Comment (bmaranville) [reply orsopy/fileio/data_source.py:91]: why would they be used as attribute names, instead of values?

**Pull request merged**